src/poc/miner_poc_statem.erl

```diff
@@ -623,28 +622,40 @@ allow_request(BlockHash, #data{blockchain=Blockchain,
                 POCInterval0
         end,
 
-    case blockchain_ledger_v1:find_gateway_info(Address, Ledger) of
-        {error, Error} ->
-            lager:warning("failed to get gateway info for ~p : ~p", [Address, Error]),
-            false;
-        {ok, GwInfo} ->
-            case blockchain:get_block(BlockHash, Blockchain) of
-                {error, Error} ->
-                    lager:warning("failed to get block ~p : ~p", [BlockHash, Error]),
-                    false;
-                {ok, Block} ->
-                    Height = blockchain_block:height(Block),
-                    case blockchain_ledger_gateway_v2:last_poc_challenge(GwInfo) of
-                        undefined ->
-                            lager:info("got block ~p @ height ~p (never challenged before)", [BlockHash, Height]),
-                            true;
-                        LastChallenge ->
-                            case (Height - LastChallenge) > POCInterval of
-                                true -> 1 == rand:uniform(trunc(POCInterval / 4));
-                                false -> false
-                            end
+    try
+        {ok, GwInfo} = blockchain_ledger_v1:find_gateway_info(Address, Ledger),
+        {ok, Block} = blockchain:get_block(BlockHash, Blockchain),
+        Height = blockchain_block:height(Block),
+        ChallengeOK =
+            case blockchain_ledger_gateway_v2:last_poc_challenge(GwInfo) of
+                undefined ->
+                    lager:info("got block ~p @ height ~p (never challenged before)", [BlockHash, Height]),
+                    true;
+                LastChallenge ->
+                    case (Height - LastChallenge) > POCInterval of
+                        true -> 1 == rand:uniform(trunc(POCInterval / 4));
+                        false -> false
                     end
-            end
+            end,
+        LocationOK = true,
+        %% the below code is tested and working.  we're going to roll
```

Comment:
You should probably also not report challenges or witnesses if you have this issue as well.

Reply:
should be remedied by a167a2d